cmd/podman/images/scp.go

```diff
@@ -0,0 +1,329 @@
+	podman := os.Args[0]
+	if v, found := os.LookupEnv("PODMAN_BINARY"); found {
+		podman = v // get path to podman bin
+	}
+	if tag != "" {
+		return "", errors.Wrapf(define.ErrInvalidArg, "Renaming of an image is currently not supported")
+	}
+	run = podman + " image save " + image + " --output=" + remoteFile // run ssh image load of the file copied via scp. Files are reverse in thie case...
+	sessions[1].Stdout = &buffer
+	sessions[1].Stderr = &bufferErr
+	if err := sessions[1].Run(run); err != nil { // run the command declared above to create new temp file
+		return "", errors.Wrapf(err, bufferErr.String())
+	}
+	n, err := scpD.CopyFrom(dial, remoteFile, imageFileIn)
+	if err != nil {
+		fmt.Println(n, " Bytes copied before error")
+		return " ", err
+	}
+	return "", nil
+}
+
+// parseLocation returns the valid connection data based off of the information provided by the user
+// iden is a string array of ssh identity keys, url is a string array of paths to ssh clients. The first entry
+// in each array is the save location and the second entry is the load location if applicable. An empty string will be
+// inserted if identity is blank (not required in some cases).
+// serv is an array of service destinations with the connection name as the index intended to be used as EngineConfig.ServiceDestinations
+// this function modifies the global scpOpt entities: FromRemote, ToRemote, Connections, and DestImageName
+func parseArgs(args []string, cfg *config.Config) ([]string, []string, map[string]config.Destination, error) {
+	iden := []string{}
+	url := []string{}
+	serv := map[string]config.Destination{}
+	cliConnections := []string{}
+	switch len(args) {
+	case 1:
+		if strings.Contains(args[0], "::") {
+			scpOpts.FromRemote = true
+			cliConnections = append(cliConnections, args[0])
+		} else {
+			scpOpts.DestImageName = args[0]
+		}
+	case 2:
+		if strings.Contains(args[0], "::") {
+			if !(strings.Contains(args[1], "::")) { // if an image is specified, this mean we are loading from our client
+				cliConnections = append(cliConnections, args[0])
+				scpOpts.ToRemote = true
+			} else if strings.Contains(args[1], "::") { // both remote clients
+				scpOpts.FromRemote = true
+				scpOpts.ToRemote = true
+				if len(strings.Split(args[0], "::")) == 1 || (len(strings.Split(args[0], "::")) == 2 && len(strings.Split(args[1], "::")) == 2) { // is save->load w/ one image name or save-load w/ two image names
+					cliConnections = append(cliConnections, args[0])
+					cliConnections = append(cliConnections, args[1])
+				} else { // else its a load save (order of args)
+					cliConnections = append(cliConnections, args[1])
+					cliConnections = append(cliConnections, args[0])
+				}
+			}
+		} else if strings.Contains(args[1], "::") && len(cliConnections) < 2 { // if we are given image host::
+			scpOpts.DestImageName = args[0]
+			scpOpts.ToRemote = true
+			cliConnections = append(cliConnections, args[1])
+		}
```

Comment:
else nothing sets `DestImageName` in the local/local case, AFAICS.

Reply:
in the local case, `DestImage` is set above by the `else` in the first case of the switch

Reply:
But the missing `else` at this line can be reached by `image scp a b`, which is also a local case.

Reply:
what does `image scp a b` mean? The only local case we support is `image scp alpine` for example since renaming does not work yet

Reply:
> what does `image scp a b` mean?

That’s a good question to ask. The consequence of asking this question should be for `parseArgs` to either define some meaning, or reject such input.

Reply:
(From other conversations, there already is code to reject `Tag != ""`; so it would be natural to parse this into (using the old/current names, `DestImageName == "a" && Tag == "b"`, and then let the execution part reject that.)

Reply:
resolved!